New Resource - `azurerm_monitor_aad_diagnostic_setting` (#11660)
terraform-providers/terraform-provider-azurerm: website/docs/r/monitor_aad_diagnostic_setting.html.markdown

  account_replication_type = "LRS"
}

locals {
  enabled_log_categories  = ["SignInLogs", "AuditLogs", "NonInteractiveUserSignInLogs", "ServicePrincipalSignInLogs"]
  disabled_log_categories = ["ManagedIdentitySignInLogs", "ProvisioningLogs", "ADFSSignInLogs"]
}

resource "azurerm_monitor_aad_diagnostic_setting" "example" {
  name               = "setting1"
  storage_account_id = azurerm_storage_account.example.id
  dynamic log {
    for_each = local.enabled_log_categories
    content {
      category = log.value
      enabled  = true
      retention_policy {
        enabled = true
        days    = 1
      }
    }
  }

  dynamic log {
    for_each = local.disabled_log_categories
    content {
      category = log.value
      enabled  = false
      retention_policy {
        enabled = false
        days    = 0
      }
    }
  }
}
```

## Arguments Reference

The following arguments are supported:

* `name` - (Required) The name which should be used for this Monitor Azure Active Directory Diagnostic Setting. Changing this forces a new Monitor Azure Active Directory Diagnostic Setting to be created.
  
* `log` - (Required) One or more `log` blocks as defined below.

> ~> **Note:** At least one of the `log` block should have the `enabled` set to `true`.

Reviewer: ```suggestion
~> **Note:** At least one of the `log` blocks must have the `enabled` property set to `true`.
```

---

* `eventhub_authorization_rule_id` - (Optional) Specifies the ID of an Event Hub Namespace Authorization Rule used to send Diagnostics Data. Changing this forces a new resource to be created.

-> **NOTE:** This can be sourced from [the `azurerm_eventhub_namespace_authorization_rule` resource](eventhub_namespace_authorization_rule.html) and is different from [a `azurerm_eventhub_authorization_rule` resource](eventhub_authorization_rule.html).

> 
> -> **NOTE:** One of `eventhub_authorization_rule_id`, `log_analytics_workspace_id` and `storage_account_id` must be specified.

Reviewer: I think it's fine to note this once after the list of properties

```suggestion
```

> * `eventhub_name` - (Optional) Specifies the name of the Event Hub where Diagnostics Data should be sent. Changing this forces a new resource to be created.
> 
> -> **NOTE:** If this isn't specified then the default Event Hub will be used.

Reviewer: This probably doesn't warrant a boxed note

```suggestion
* `eventhub_name` - (Optional) Specifies the name of the Event Hub where Diagnostics Data should be sent. If not specified, the default Event Hub will be used. Changing this forces a new resource to be created.
```

* `log_analytics_workspace_id` - (Optional) Specifies the ID of a Log Analytics Workspace where Diagnostics Data should be sent.

> 
> -> **NOTE:** One of `eventhub_authorization_rule_id`, `log_analytics_workspace_id` and `storage_account_id` must be specified.

Reviewer: As above

```suggestion
```
